packages/block-editor/src/components/block-list/style.scss

```diff
@@ -138,13 +138,15 @@
 	}
 
 	// Spotlight mode.
-	&.is-focus-mode:not(.is-multi-selected) {
+	.is-focus-mode &:not(.is-multi-selected) {
 		opacity: 0.5;
 		transition: opacity 0.1s linear;
 		@include reduce-motion("transition");
 
-		&:not(.is-focused) .block-editor-block-list__block,
-		&.is-focused {
+		&:not(.is-selected) .block-editor-block-list__block,
+		&.has-child-selected,
+		&.is-typing,
```

Comment:
This is the same fix as

https://github.com/WordPress/gutenberg/blob/17e5c2d870d84fb6de48dcd5bc3c38cd0c0fb0d0/packages/block-library/src/button/editor.scss#L68-L69

Ideally, `is-selected` should always be applied, and there should be a separate class for showing/hiding selection styles (appearing selected).